src/views/News/index.js

```diff
@@ -146,31 +256,205 @@ export default class StudentNews extends Component {
+    return new Promise((resolve, reject) => {
+      clearTimeout(this.photoDialogErrorTimeout);
+      this.setState({ photoDialogErrorTimeout: null, photoDialogError: null });
+      resolve(true);
+    });
+  }
+
+  /**
+   * Creates the Photo Dialog message that will be displayed to the user
+   *
+   * @return {String} The message of the Photo Dialog
+   */
+  createPhotoDialogBoxMessage() {
+    let message = '';
+    // If an error occured and there's no currently running timeout, the error is displayed
+    // and a timeout for that error message is created
+    if (this.state.photoDialogError !== null) {
+      message = <span style={{ color: '#B63228' }}>{this.state.photoDialogError}</span>;
+      if (this.state.photoDialogErrorTimeout === null) {
+        // Shows the error message for 6 seconds and then returns back to normal text
+        this.setState({
+          photoDialogErrorTimeout: setTimeout(() => {
+            this.setState({ photoDialogErrorTimeout: null }); //photoDialogErrorTimeout = null;
+            this.setPhotoDialogError(null);
+          }, 6000),
+        });
+      }
+    }
+
+    // If no error occured and the cropper is shown, the cropper text is displayed
+    else if (this.state.showCropper && !this.state.justShowPicture) {
+      message = 'Crop Photo to liking & Click Submit';
+    }
+
+    // If no error occured and the edit window has just been opened,
+    // show previously submitted photo
+    else if (this.state.justShowPicture) {
+      message = 'Previously submitted photo';
+    }
+
+    // If no error occured and the cropper is not shown, the pick a file text is displayed
+    else {
+      message =
+        this.breakpointWidth === 'md' ||
+        this.breakpointWidth === 'sm' ||
+        this.breakpointWidth === 'xs'
+          ? //currentWidth === 'md' || currentWidth === 'sm' || currentWidth === 'xs'
+            'Tap Image to Browse Files'
+          : 'Drag & Drop Picture, or Click to Browse Files';
+    }
+    return message;
+  }
+
+  // Method called when 'edit' clicked for a news item
   async handleNewsItemEdit(newsID) {
     let newsItem = await newsService.getPostingByID(newsID);
+
     this.setState({
       openPostActivity: true,
       newPostCategory: newsItem.categoryID,
       newPostSubject: newsItem.Subject,
       newPostBody: newsItem.Body,
       currentlyEditing: newsID,
     });
+
+    /*
+    Error checking. Theoretically, this code is designed so that
+    When the get method in the API service is called from the frontend,
+    it will return the image data, even though that's not the value
+    of the image column in news entries. But in the impossible event that
+    it somhow DID return the path of the image instead of the image data,
+    not only would that produce garbage and make the cropper have trouble,
+    but it also is a potential security concern; it sends data back to the
+    client that shouldn't be sent anywhere.
```

Comment:
I should add that that would also be a security issue; if the path were returned it would include the SNID of the poster.

Reply:
`SNID` is an arbitrary number that uniquely identifies student news postings, so there's no issue with exposing it to the front end whatsoever. I think maybe you're confusing it with `ID_NUM`, the user's ID number?

Reply:
Oh- yes, I am.